### PR TITLE
WIP: Docker pass env node

### PR DESCRIPTION
**Pull request by AntonEmery**

### CODE REVIEW

File: src/server/start.js [hunk 1]
```diff
@@ -1,8 +1,5 @@
 const mongoose = require('mongoose');
 
-// Import variables.env file
```
Comment (XiXiaPdx) [inline src/server/start.js:3]: This was originally here so the server recognized where to find the .env file? 

So if this is gone, how does the server know where to pick up the .env file and the variables in it?

File: src/server/start.js [hunk 2]
```diff
@@ -1,8 +1,5 @@
...
-require('dotenv').config({ path: 'src/server/variables.env' });
-
```
Comment (XiXiaPdx) [inline src/server/start.js:5]: This was originally here so the server recognized where to find the .env file?

So if this is gone, how does the server know where to pick up the .env file and the variables in it?

Comment (XiXiaPdx) [reply src/server/start.js:5]: Ok, I see now. The variables are being referenced by `process.env.nameOfVariable` in the app. 

We are adding those variables to the environment through docker run --env-file  _filename_

Comment (AntonEmery) [reply src/server/start.js:5]: In the `Makefile` the `--env-file variables.env` makes that file available to Node on `process.env`

https://github.com/AntonEmery/chord-app/blob/9884d8d2b83cc143436457b8f803383c2908ddd3/Makefile#L8

**Comment by XiXiaPdx** [comment]
I think if you can add a .dockerignore, this will be a good chunck of work!

**Comment by AntonEmery** [comment]
@XiXiaPdx Yea if i could get the `.dockerignore` file to work i'd merge this into Master.  I added it, but I think due to the volume mounting  from my local machine to the container, so `nodemon` can work, it copies over everything.  `v ${PWD}:/chord-app`  If I remove that bit the .dockerignore file seems to work.  

https://github.com/AntonEmery/chord-app/blob/bf0036b6a5ea84a62bdd0d1dd2d20995ddbe90b4/Makefile#L8

File: Makefile [hunk 3]
```diff
@@ -10,7 +10,7 @@ node_dev:
 node_prod:
 	echo 'Starting Node prod server in Docker Container'
 	docker build --build-arg run_mode_arg=start -t node_prod .
-	docker run -p 8080:8080 -e "run_mode_env=start" node_prod
+	docker run -p 8080:8080 -e "run_mode_env=start" node_prod sh
```
Comment (XiXiaPdx) [inline Makefile:13]: Did you want the `sh`  here to be committed?

Comment (AntonEmery) [reply Makefile:13]: No, I should probably get rid of it before I merge. 

This article was useful.  https://www.freecodecamp.org/news/how-to-enable-live-reload-on-docker-based-applications/

But the ignored files still appear when I remote into the container.  But i don't know if they are actually in the container or a result of the mirroring

File: .dockerignore [hunk 4]
```diff
@@ -0,0 +1,4 @@
+node_modules
+src/client
+variables.env
+notes.txt
```
Comment (XiXiaPdx) [inline .dockerignore:4]: Can you also add the following? 
.idea
.git

**Comment by AntonEmery** [comment]
Closes #31

**Pull request merged**